src/services/mentor/mentor-router.ts

```diff
@@ -0,0 +1,234 @@
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "InvalidPermission"));
+    }
+
+    const officeHours: OfficeHoursFormat | null = await Models.MentorOfficeHours.findOne({ mentorId: mentorId });
+
+    if (!officeHours) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "MentorNotFound"));
+    }
+
+    await Models.MentorOfficeHours.findOneAndDelete({ mentorId: mentorId });
+
+    return res.status(StatusCode.SuccessOK).send("Success");
+});
+
+/**
+ * @api {post} /mentor/attendance POST /mentor/attendance
+ * @apiGroup Mentor
+ * @apiDescription Checks an attendee into a mentor's office hours.
+ *
+ * @apiBody {String} mentorId id of the mentor to delete
+ * @apiParamExample {json} Example Request:
+ * { "mentorId": "Jojos" }
+ *
+ *
+ * @apiSuccess (200: Success) {String} success Indicates successful checkin.
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ *   "Success"
+ *
+ * @apiError (400: Invalid Request) {String} InvalidRequest
+ * @apiErrorExample Example Error Response (InvalidRequest):
+ *     HTTP/1.1 400 Invalid Request
+ *     {"error": "InvalidRequest"}
+ *
+ * @apiError (403: Invalid Permission) {String} InvalidPermission Caller has invalid permissions.
+ * @apiErrorExample Example Error Response (InvalidPermission):
+ *     HTTP/1.1 403 Invalid Permission
+ *     {"error": "InvalidPermission"}
+ *
+ * @apiError (400: Invalid Request) {String} MentorNotFound
+ * @apiErrorExample Example Error Response (InvalidPermission):
+ *     HTTP/1.1 403 Invalid Request
+ *     {"error": "MentorNotFound"}
+ *
+ * @apiError (400: Invalid Request) {String} AlreadyCheckedIn
+ * @apiErrorExample Example Error Response (InvalidPermission):
+ *     HTTP/1.1 403 Invalid Request
+ *     {"error": "AlreadyCheckedIn"}
+ */
+mentorRouter.post("/attendance/", strongJwtVerification, async (req: Request, res: Response, next: NextFunction) => {
+    const mentorId: string | undefined = req.body.mentorId;
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+
+    if (!mentorId) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "InvalidRequest"));
+    }
+    // Checks that the caller is an attendee
+    if (hasAdminPerms(payload) || hasStaffPerms(payload)) {
+        return next(new RouterError(StatusCode.ClientErrorForbidden, "InvalidPermission"));
+    }
+
+    const officeHours: OfficeHoursFormat | null = await Models.MentorOfficeHours.findOne({ mentorId: mentorId });
+
+    if (!officeHours) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "MentorNotFound"));
+    }
+
+    // Checks whether the attendee has already checked in for the office hours
+    if (officeHours.attendees.includes(payload.id)) {
+        return next(new RouterError(StatusCode.ClientErrorBadRequest, "AlreadyCheckedIn"));
+    }
+
+    await updatePoints(payload.id, pointCoinUpdateValue);
+    await updateCoins(payload.id, pointCoinUpdateValue);
```

Comment:
Can you validate that user sending request is non-staff (`ATTENDEE`) role (check JWT), or they won't have a profile. Ideally we catch and error here, instead of in these functions. Deferring to @AydanPirani to confirm